src/parser/shared/modules/items/bfa/raids/battleofdazaralor/WardOfEnvelopment.js

```diff
@@ -0,0 +1,154 @@
+        },
+      });
+    }
+  }
+
+  onCast(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_PROTECTION.id){
+      this.uses += 1;
+      this.firstExpiration = true; // Reset it back to the first expiration
+    }
+  }
+
+  onAbsorb(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_PROTECTION.id){
+      this.absorbUsed += (event.amount || 0) + (event.absorbed || 0);
+    }
+  }
+
+  onBuff(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_PROTECTION.id){
+      this.targetsHit += 1;
+    }
+  }
+
+  onRemoveBuff(event) {
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.ENVELOPING_PROTECTION.id){
+      if (this.firstExpiration === true) { // If this is the first buff expiration for this cast, add it to the total wasted
+        this.absorbWasted += event.absorb;
+        this.firstExpiration = false; // Set it to false so it doesn't add the absorb for every other buff
+      }
+    }
+  }
+
+  get possibleUseCount() {
+    return 1 + Math.floor(this.owner.fightDuration / (ACTIVATION_COOLDOWN * 1000));
+  }
+
+  // Average amount of damage absorbed on each cast
+  get averageAbsorbPerCast() {
+    return this.absorbUsed / this.uses;
+  }
+
+  // Total amount of absorbs put out
+  get totalAbsorb() {
+    return this.absorbUsed + this.absorbWasted;
+  }
+
+  // Percentage of absorbs put out that was wasted by expiring
+  get wastedPercentage() {
+    return this.absorbWasted / (this.absorbUsed + this.absorbWasted);
+  }
+
+  // Average number of allies hit (max of 5)
+  get averageTargetsHit() {
+    return this.targetsHit / this.uses;
+  }
+
+  // Extra shield value gained by hitting multiple allies (up to 5, 7.5% for each after base)
+  get gainedShieldValue() {
+    return Math.ceil((this.targetsHit / this.uses) - 1) * 0.075;
+  }
+  
+  item() {
+    return {
+      item: ITEMS.WARD_OF_ENVELOPMENT,
+      result: (
+        <dfn data-tip={`
+          You activated your Ward of Envelopment <b>${this.uses}</b> of <b>${this.possibleUseCount}</b> possible time${this.uses === 1 ? '' : 's'} with an average of <b>${formatNumber(this.averageAbsorbPerCast)}</b> absorption per use.
+          It absorbed <b>${formatNumber(this.absorbUsed)}</b> out of <b>${formatNumber(this.totalAbsorb)}</b> damage and <b>${formatNumber(this.absorbWasted)} (${formatPercentage(this.wastedPercentage)}%)</b> was unused. <br />
+          On average you hit <b>${formatNumber(this.averageTargetsHit)}</b> out of <b>5</b> allies and gained <b>${formatPercentage(this.gainedShieldValue)}%</b> extra absorption value. <br />
+          `}>
+          <ItemHealingDone amount={this.absorbUsed} />
+        </dfn>
+      ),
+    };
+  }
+
+  get suggestedShieldValue() {
+    return {
+      actual: this.averageTargetsHit,
+      isLessThan: {
+        minor: 5,
+        average: 4,
+        major: 3,
+      },
+      style: 'percentage',
```

Comment:
```suggestion
      style: 'number',
```
will probably be better